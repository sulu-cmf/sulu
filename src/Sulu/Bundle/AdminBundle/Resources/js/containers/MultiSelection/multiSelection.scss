@import '../../containers/Application/colors.scss';

$multiSelectionMainColumnColor: $black;
$multiSelectionColumnColor: $scorpion;

.item-column {
    color: $multiSelectionColumnColor;
    display: inline-block;

    &:first-child {
        color: $multiSelectionMainColumnColor;
    }
}
