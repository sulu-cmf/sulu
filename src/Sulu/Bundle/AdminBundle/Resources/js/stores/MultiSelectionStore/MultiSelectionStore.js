// @flow
import {action, observable} from 'mobx';
import type {IObservableValue} from 'mobx';
import {arrayMove} from '../../components';
import {ResourceRequester} from '../../services';

export default class SelectionStore {
    @observable items: Array<Object> = [];
    @observable loading: boolean = false;
    resourceKey: string;
    locale: ?IObservableValue<string>;
    idFilterParameter: string;

    constructor(
        resourceKey: string,
        selectedItemIds: Array<string | number>,
        locale: ?IObservableValue<string>,
        idFilterParameter: string = 'ids'
    ) {
        this.resourceKey = resourceKey;
        this.locale = locale;
        this.idFilterParameter = idFilterParameter;
        if (selectedItemIds.length) {
            this.loadItems(selectedItemIds);
        }
    }

    @action set(items: Array<Object>) {
        this.items = items;
    }

    @action removeById(id: string | number) {
        // TODO use metadata instead of hardcoded id
        this.items.splice(this.items.findIndex((item) => item.id === id), 1);
    }

    @action move(oldItemIndex: number, newItemIndex: number) {
        this.items = arrayMove(this.items, oldItemIndex, newItemIndex);
    }

    @action setLoading(loading: boolean) {
        this.loading = loading;
    }

    @action loadItems = (itemIds: ?Array<string | number>) => {
        if (!itemIds || itemIds.length === 0) {
            this.items = [];
            return;
        }

        this.setLoading(true);
        return ResourceRequester.getList(this.resourceKey, {
            locale: this.locale ? this.locale.get() : undefined,
            [this.idFilterParameter]: itemIds.join(','),
            limit: undefined,
            page: 1,
        }).then(action((data) => {
            this.items = data._embedded[this.resourceKey];
            this.setLoading(false);
        }));
    };
}
