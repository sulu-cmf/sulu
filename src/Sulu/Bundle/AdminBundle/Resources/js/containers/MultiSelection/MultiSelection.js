// @flow
import React, {Fragment} from 'react';
import {action, autorun, observable, toJS} from 'mobx';
import type {IObservableValue} from 'mobx';
import {observer} from 'mobx-react';
import equals from 'fast-deep-equal';
import {MultiItemSelection} from '../../components';
import MultiSelectionStore from '../../stores/MultiSelectionStore';
import MultiDatagridOverlay from '../MultiDatagridOverlay';
import multiSelectionStyles from './multiSelection.scss';

type Props = {|
    adapter: string,
    disabled: boolean,
    disabledIds: Array<string | number>,
    displayProperties: Array<string>,
    onChange: (selectedIds: Array<string | number>) => void,
    label?: string,
    locale?: ?IObservableValue<string>,
    icon: string,
    resourceKey: string,
    value: Array<string | number>,
    overlayTitle: string,
|};

@observer
export default class MultiSelection extends React.Component<Props> {
    static defaultProps = {
        disabled: false,
        disabledIds: [],
        displayProperties: [],
        icon: 'su-plus',
        value: [],
    };

    selectionStore: MultiSelectionStore;
    changeDisposer: () => void;
    changeAutorunInitialized: boolean = false;

    @observable overlayOpen: boolean = false;

    constructor(props: Props) {
        super(props);

        const {onChange, locale, resourceKey, value} = this.props;

        this.selectionStore = new MultiSelectionStore(resourceKey, value, locale);
        this.changeDisposer = autorun(() => {
            const {value} = this.props;
            const itemIds = this.selectionStore.items.map((item) => item.id);

            if (!this.changeAutorunInitialized) {
                this.changeAutorunInitialized = true;
                return;
            }

            if (equals(toJS(value), toJS(itemIds))) {
                return;
            }

            onChange(itemIds);
        });
    }

    componentDidUpdate() {
        const newValue = toJS(this.props.value);
        const oldValue = toJS(this.selectionStore.items.map((item) => item.id));

        newValue.sort();
        oldValue.sort();
        if (!equals(newValue, oldValue) && !this.selectionStore.loading) {
            this.selectionStore.loadItems(newValue);
        }
    }

    componentWillUnmount() {
        this.changeDisposer();
    }

    @action closeOverlay() {
        this.overlayOpen = false;
    }

    @action openOverlay() {
        this.overlayOpen = true;
    }

    @action handleOverlayOpen = () => {
        this.openOverlay();
    };

    @action handleOverlayClose = () => {
        this.closeOverlay();
    };

    handleOverlayConfirm = (selectedItems: Array<Object>) => {
        this.selectionStore.set(selectedItems);
        this.closeOverlay();
    };

    handleRemove = (id: number | string) => {
        this.selectionStore.removeById(id);
    };

    handleSorted = (oldItemIndex: number, newItemIndex: number) => {
        this.selectionStore.move(oldItemIndex, newItemIndex);
    };

    render() {
        const {
            adapter,
            disabled,
            disabledIds,
            displayProperties,
            icon,
            label,
            locale,
            resourceKey,
            overlayTitle,
        } = this.props;

        const {items, loading} = this.selectionStore;
        const columns = displayProperties.length;

        return (
            <Fragment>
                <MultiItemSelection
                    disabled={disabled}
                    label={label}
                    leftButton={{
                        icon,
                        onClick: this.handleOverlayOpen,
                    }}
                    loading={loading}
                    onItemRemove={this.handleRemove}
                    onItemsSorted={this.handleSorted}
                >
                    {items.map((item, index) => (
                        <MultiItemSelection.Item id={item.id} index={index + 1} key={item.id}>
                            <div>
                                {displayProperties.map((displayProperty) => (
                                    <span
                                        className={multiSelectionStyles.itemColumn}
                                        key={displayProperty}
                                        style={{width: 100 / columns + '%'}}
                                    >
                                        {item[displayProperty]}
                                    </span>
                                ))}
                            </div>
                        </MultiItemSelection.Item>
                    ))}
                </MultiItemSelection>
                <MultiDatagridOverlay
                    adapter={adapter}
                    disabledIds={disabledIds}
                    locale={locale}
                    onClose={this.handleOverlayClose}
                    onConfirm={this.handleOverlayConfirm}
                    open={this.overlayOpen}
                    preSelectedItems={items}
                    resourceKey={resourceKey}
                    title={overlayTitle}
                />
            </Fragment>
        );
    }
}
