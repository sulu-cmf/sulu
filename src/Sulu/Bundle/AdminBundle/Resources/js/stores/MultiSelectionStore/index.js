// @flow
import MultiSelectionStore from './MultiSelectionStore';

export default MultiSelectionStore;
