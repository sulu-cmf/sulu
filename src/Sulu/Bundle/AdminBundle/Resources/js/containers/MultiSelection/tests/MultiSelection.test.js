// @flow
import React from 'react';
import {mount, render, shallow} from 'enzyme';
import {observable} from 'mobx';
import pretty from 'pretty';
import MultiSelection from '../MultiSelection';
import MultiSelectionStore from '../../../stores/MultiSelectionStore';

jest.mock('../../../utils', () => ({
    translate: jest.fn((key) => key),
}));

jest.mock('../../../containers/Datagrid', () => function Datagrid() {
    return <div className="datagrid" />;
});

jest.mock('../../../containers/Datagrid/stores/DatagridStore', () => jest.fn(function(resourceKey) {
    this.clearSelection = jest.fn();
    this.destroy = jest.fn();
    this.resourceKey = resourceKey;
    this.select = jest.fn();
    this.setActive = jest.fn();
}));

jest.mock('../../../stores/MultiSelectionStore', () => jest.fn(function() {
    this.items = [];
    this.set = jest.fn();
    this.move = jest.fn();
    this.removeById = jest.fn();
    this.loadItems = jest.fn();
}));

beforeEach(() => {
    const body = document.body;

    if (body) {
        body.innerHTML = '';
    }
});

test('Show with default plus icon', () => {
    expect(render(
        <MultiSelection adapter="table" onChange={jest.fn()} overlayTitle="Selection" resourceKey="snippets" />
    )).toMatchSnapshot();
});

test('Show with passed label', () => {
    expect(render(
        <MultiSelection
            adapter="column_list"
            label="Select Snippets"
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="snippets"
        />
    )).toMatchSnapshot();
});

test('Show with passed icon', () => {
    expect(render(
        <MultiSelection
            adapter="table"
            icon="su-document"
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="snippets"
        />
    )).toMatchSnapshot();
});

test('Pass locale to MultiDatagridOverlay', () => {
    const locale = observable.box('de');
    const selection = mount(
        <MultiSelection
            adapter="table"
            locale={locale}
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="snippets"
        />
    );

    expect(selection.find('MultiDatagridOverlay').prop('locale').get()).toEqual('de');
});

test('Pass disabledIds to MultiDatagridOverlay', () => {
    const disabledIds = [1, 2, 4];

    const selection = mount(
        <MultiSelection
            adapter="table"
            disabledIds={disabledIds}
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="snippets"
        />
    );

    expect(selection.find('MultiDatagridOverlay').prop('disabledIds')).toEqual(disabledIds);
});

test('Show with passed values as items in right locale', () => {
    const locale = observable.box('en');

    // $FlowFixMe
    MultiSelectionStore.mockImplementationOnce(function() {
        this.items = [{id: 1, title: 'Title 1'}, {id: 2, title: 'Title 2'}, {id: 5, title: 'Title 5'}];
    });

    expect(render(
        <MultiSelection
            adapter="table"
            displayProperties={['id', 'title']}
            locale={locale}
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="snippets"
            value={[1, 2, 5]}
        />
    )).toMatchSnapshot();

    expect(MultiSelectionStore).toBeCalledWith('snippets', [1, 2, 5], locale);
});

test('Should open an overlay', () => {
    const selection = mount(
        <MultiSelection adapter="table" onChange={jest.fn()} overlayTitle="Selection" resourceKey="snippets" />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');

    const body = document.body;
    expect(pretty(body ? body.innerHTML : null)).toMatchSnapshot();
});

test('Should close an overlay using the close button', () => {
    const selection = mount(
        <MultiSelection
            adapter="table"
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="snippets"
        />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');

    const closeButton = document.querySelector('.su-times');
    if (closeButton) {
        closeButton.click();
    }

    selection.update();
    expect(selection.find('MultiDatagridOverlay').prop('open')).toEqual(false);
});

test('Should close an overlay using the confirm button', () => {
    const selection = mount(
        <MultiSelection adapter="table" onChange={jest.fn()} overlayTitle="Selection" resourceKey="snippets" />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');

    const confirmButton = document.querySelector('button.primary');
    if (confirmButton) {
        confirmButton.click();
    }

    selection.update();
    expect(selection.find('MultiDatagridOverlay').prop('open')).toEqual(false);
});

test('Should call the onChange callback when clicking the confirm button', () => {
    const changeSpy = jest.fn();
    const selection = mount(
        <MultiSelection adapter="table" onChange={changeSpy} overlayTitle="Selection" resourceKey="snippets" />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');
    const datagridStore = selection.find('MultiDatagridOverlay').instance().datagridStore;
    datagridStore.selections = [3, 7, 2];

    const confirmButton = document.querySelector('button.primary');
    if (confirmButton) {
        confirmButton.click();
    }

    expect(selection.instance().selectionStore.set).toBeCalledWith([3, 7, 2]);
});

test('Should not call the onChange callback when items have not changed', () => {
    const changeSpy = jest.fn();
    const selection = mount(
        <MultiSelection
            adapter="table"
            onChange={changeSpy}
            overlayTitle="Selection"
            resourceKey="snippets"
            value={[1]}
        />
    );

    expect(changeSpy).not.toBeCalled();

    selection.instance().selectionStore.items = [{id: 1}];
    selection.setProps({value: [1]});
    expect(changeSpy).not.toBeCalled();
});

test('Should load the items if value prop changes', () => {
    const changeSpy = jest.fn();
    const selection = mount(
        <MultiSelection
            adapter="table"
            onChange={changeSpy}
            overlayTitle="Selection"
            resourceKey="snippets"
            value={[1]}
        />
    );

    selection.setProps({value: [1, 3]});
    expect(selection.instance().selectionStore.loadItems).toBeCalledWith([1, 3]);
});

test('Should instantiate the DatagridStore with the correct resourceKey and destroy it on unmount', () => {
    const selection = mount(
        <MultiSelection adapter="table" onChange={jest.fn()} overlayTitle="Selection" resourceKey="pages" />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');

    const datagridStore = selection.find('MultiDatagridOverlay').instance().datagridStore;
    expect(datagridStore.resourceKey).toEqual('pages');

    selection.unmount();
    expect(datagridStore.destroy).toBeCalled();
});

test('Should instantiate the DatagridStore with the preselected ids', () => {
    // $FlowFixMe
    MultiSelectionStore.mockImplementationOnce(function() {
        this.items = [{id: 1}, {id: 5}, {id: 8}];
    });

    const selection = mount(
        <MultiSelection
            adapter="table"
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="pages"
            value={[1, 5, 8]}
        />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');

    const datagridStore = selection.find('MultiDatagridOverlay').instance().datagridStore;
    expect(datagridStore.select).toBeCalledWith({id: 1});
    expect(datagridStore.select).toBeCalledWith({id: 5});
    expect(datagridStore.select).toBeCalledWith({id: 8});
});

test('Should reinstantiate the DatagridStore with the preselected ids when new props are received', () => {
    const locale = observable.box('en');

    // $FlowFixMe
    MultiSelectionStore.mockImplementationOnce(function() {
        this.items = [{id: 1}, {id: 5}, {id: 8}];
        this.loadItems = jest.fn();
    });

    const selection = mount(
        <MultiSelection
            adapter="table"
            locale={locale}
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="pages"
            value={[1, 5, 8]}
        />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');

    const datagridStore = selection.find('MultiDatagridOverlay').instance().datagridStore;
    expect(datagridStore.select).toBeCalledWith({id: 1});
    expect(datagridStore.select).toBeCalledWith({id: 5});
    expect(datagridStore.select).toBeCalledWith({id: 8});

    selection.setProps({value: [1, 3]});
    const loadItemsCall = selection.instance().selectionStore.loadItems.mock.calls[0];
    expect(loadItemsCall[0]).toEqual([1, 3]);
});

test('Should not reload items if none of the items changed', () => {
    const locale = observable.box('en');

    // $FlowFixMe
    MultiSelectionStore.mockImplementationOnce(function() {
        this.items = [{id: 1}, {id: 5}, {id: 8}];
        this.loadItems = jest.fn();
    });

    const selection = mount(
        <MultiSelection
            adapter="table"
            locale={locale}
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="pages"
            value={[1, 5, 8]}
        />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');

    const datagridStore = selection.find('MultiDatagridOverlay').instance().datagridStore;
    expect(datagridStore.select).toBeCalledWith({id: 1});
    expect(datagridStore.select).toBeCalledWith({id: 5});
    expect(datagridStore.select).toBeCalledWith({id: 8});

    selection.setProps({value: [1, 5, 8]});
    expect(selection.instance().selectionStore.loadItems).not.toBeCalled();
});

test('Should not reinstantiate the DatagridStore with the preselected ids when new props have the same values', () => {
    const locale = observable.box('en');

    // $FlowFixMe
    MultiSelectionStore.mockImplementationOnce(function() {
        this.items = [{id: 1}, {id: 5}, {id: 8}];
        this.loadItems = jest.fn();
    });

    const selection = mount(
        <MultiSelection
            adapter="table"
            locale={locale}
            onChange={jest.fn()}
            overlayTitle="Selection"
            resourceKey="pages"
            value={[1, 5, 8]}
        />
    );

    selection.find('Button[icon="su-plus"]').simulate('click');

    selection.setProps({value: [1, 5, 8]});
    expect(selection.instance().selectionStore.loadItems).not.toBeCalled();
});

test('Should remove an item when the remove button is clicked', () => {
    const changeSpy = jest.fn();
    const selection = shallow(
        <MultiSelection
            adapter="table"
            onChange={changeSpy}
            overlayTitle="Selection"
            resourceKey="snippets"
            value={[3, 7, 9]}
        />
    );

    selection.find('MultiItemSelection').prop('onItemRemove')(7);
    expect(selection.instance().selectionStore.removeById).toBeCalledWith(7);
});

test('Should reorder the items on drag and drop', () => {
    const changeSpy = jest.fn();
    const selection = shallow(
        <MultiSelection
            adapter="table"
            onChange={changeSpy}
            overlayTitle="Selection"
            resourceKey="snippets"
            value={[3, 7, 9]}
        />
    );

    selection.find('MultiItemSelection').prop('onItemsSorted')(1, 2);

    expect(selection.instance().selectionStore.move).toBeCalledWith(1, 2);
});
