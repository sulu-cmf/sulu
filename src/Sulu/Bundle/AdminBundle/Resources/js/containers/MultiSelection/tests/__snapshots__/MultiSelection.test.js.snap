// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show with passed icon and label and open overlay 1`] = `
Array [
  <div
    class="multiItemSelectionClass"
  >
    <div
      class="header"
    >
      <button
        class="button left"
        type="button"
      >
        <span
          aria-label="su-document"
          class="su-document icon"
        />
      </button>
      <div
        class="label"
      >
        Select Snippets
      </div>
    </div>
    <ul
      class="list"
    >
      <li
        class="listElement"
      >
        <div
          class="item"
        >
          <span
            class="dragHandle sortable"
          >
            <span
              aria-label="su-more"
              class="su-more"
            />
            <span
              class="index"
            >
              1
            </span>
          </span>
          <div
            class="content"
          >
            <div>
              <span
                class="itemColumn"
                style="width: 50%;"
              >
                <div
                  aria-label="1"
                  class="croppedText"
                  title="1"
                >
                  <div
                    aria-hidden="true"
                    class="front"
                  >
                    1
                  </div>
                  <div
                    aria-hidden="true"
                    class="back"
                  >
                    <span />
                  </div>
                  <div
                    class="whole"
                  >
                    1
                  </div>
                </div>
              </span>
              <span
                class="itemColumn"
                style="width: 50%;"
              >
                <div
                  aria-label="Title 1"
                  class="croppedText"
                  title="Title 1"
                >
                  <div
                    aria-hidden="true"
                    class="front"
                  >
                    Titl
                  </div>
                  <div
                    aria-hidden="true"
                    class="back"
                  >
                    <span>
                      e 1
                    </span>
                  </div>
                  <div
                    class="whole"
                  >
                    Title 1
                  </div>
                </div>
              </span>
            </div>
          </div>
          <div
            class="buttons"
          >
            <button
              class="button"
              type="button"
            >
              <span
                aria-label="su-trash-alt"
                class="su-trash-alt"
              />
            </button>
          </div>
        </div>
      </li>
      <li
        class="listElement"
      >
        <div
          class="item"
        >
          <span
            class="dragHandle sortable"
          >
            <span
              aria-label="su-more"
              class="su-more"
            />
            <span
              class="index"
            >
              2
            </span>
          </span>
          <div
            class="content"
          >
            <div>
              <span
                class="itemColumn"
                style="width: 50%;"
              >
                <div
                  aria-label="2"
                  class="croppedText"
                  title="2"
                >
                  <div
                    aria-hidden="true"
                    class="front"
                  >
                    2
                  </div>
                  <div
                    aria-hidden="true"
                    class="back"
                  >
                    <span />
                  </div>
                  <div
                    class="whole"
                  >
                    2
                  </div>
                </div>
              </span>
              <span
                class="itemColumn"
                style="width: 50%;"
              >
                <div
                  aria-label="Title 2"
                  class="croppedText"
                  title="Title 2"
                >
                  <div
                    aria-hidden="true"
                    class="front"
                  >
                    Titl
                  </div>
                  <div
                    aria-hidden="true"
                    class="back"
                  >
                    <span>
                      e 2
                    </span>
                  </div>
                  <div
                    class="whole"
                  >
                    Title 2
                  </div>
                </div>
              </span>
            </div>
          </div>
          <div
            class="buttons"
          >
            <button
              class="button"
              type="button"
            >
              <span
                aria-label="su-trash-alt"
                class="su-trash-alt"
              />
            </button>
          </div>
        </div>
      </li>
      <li
        class="listElement"
      >
        <div
          class="item"
        >
          <span
            class="dragHandle sortable"
          >
            <span
              aria-label="su-more"
              class="su-more"
            />
            <span
              class="index"
            >
              3
            </span>
          </span>
          <div
            class="content"
          >
            <div>
              <span
                class="itemColumn"
                style="width: 50%;"
              >
                <div
                  aria-label="5"
                  class="croppedText"
                  title="5"
                >
                  <div
                    aria-hidden="true"
                    class="front"
                  >
                    5
                  </div>
                  <div
                    aria-hidden="true"
                    class="back"
                  >
                    <span />
                  </div>
                  <div
                    class="whole"
                  >
                    5
                  </div>
                </div>
              </span>
              <span
                class="itemColumn"
                style="width: 50%;"
              >
                <div
                  aria-label="Title 5"
                  class="croppedText"
                  title="Title 5"
                >
                  <div
                    aria-hidden="true"
                    class="front"
                  >
                    Titl
                  </div>
                  <div
                    aria-hidden="true"
                    class="back"
                  >
                    <span>
                      e 5
                    </span>
                  </div>
                  <div
                    class="whole"
                  >
                    Title 5
                  </div>
                </div>
              </span>
            </div>
          </div>
          <div
            class="buttons"
          >
            <button
              class="button"
              type="button"
            >
              <span
                aria-label="su-trash-alt"
                class="su-trash-alt"
              />
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  <div
    class="backdrop visible fixed"
    role="button"
  />,
  <div
    class="container"
  >
    <div
      class="overlay large"
    >
      <section
        class="content"
      >
        <header>
          <h2>
            Selection
          </h2>
          <span
            aria-label="su-times"
            class="su-times clickable icon"
            role="button"
            tabindex="0"
          />
        </header>
        <article>
          <div
            class="adapterContainer overlay table"
          >
            <div
              class="list adapter table"
            >
              <div
                class="list"
              />
            </div>
          </div>
        </article>
        <footer>
          <button
            class="button primary"
            type="button"
          >
            <span
              class="text"
            >
              sulu_admin.confirm
            </span>
          </button>
        </footer>
      </section>
    </div>
  </div>,
]
`;
